packages/protocol/contracts/tokenvault/BridgedERC20Base.sol

```diff
@@ -82,14 +83,30 @@ abstract contract BridgedERC20Base is EssentialContract, IBridgedERC20 {
             emit MigratedTo(migratingAddress, _account, _amount);
             // Ask the new bridged token to mint token for the user.
             IBridgedERC20(migratingAddress).mint(_account, _amount);
-        } else if (msg.sender != resolve(LibStrings.B_ERC20_VAULT, true)) {
-            // Only the vault can burn tokens when not migrating out
-            revert RESOLVER_DENIED();
+        } else {
+            // When user wants to burn tokens only during 'migrating out' phase is possible. If
```

Comment:
So the `ERC20Vault` was allowed to `burn` - but `burn` only happened during bridging back the 'bridged' tokens to canonical. Now, since it is using the other `burn(amount)` function, i dont think we need this code section.

Reply:
But i can delete the `oldBurn` and incorporate this into the `newBurn`